lib/AST/Type.cpp

```diff
@@ -2027,9 +2027,14 @@ static ForeignRepresentableKind
 getObjCObjectRepresentable(Type type, const DeclContext *dc) {
   // @objc metatypes are representable when their instance type is.
   if (auto metatype = type->getAs<AnyMetatypeType>()) {
+    auto instanceType = metatype->getInstanceType();
+
+    // Consider protocol metatypes as not Objective-C representable.
+    if (metatype->is<MetatypeType>() && instanceType->is<ProtocolType>())
+      return ForeignRepresentableKind::None;
```

Comment:
Well, protocol compositions (id<P,Q>) and existential metatypes (Class<P>) should be ObjC-representable if their protocols are all @objc, so if that's the condition, they'd get by.